frontend/src/static/js/v_zoom.js

```diff
@@ -10,8 +14,19 @@ window.vZoom = {
     getSliceLink(slice) {
       return `${window.getBaseLink(this.info.user.repoId)}/commit/${slice.hash}`;
     },
+
+    getFilterTimeFrame() {
+      window.decodeHash();
+      const hash = window.hashParams;
+      if (hash.timeframe) {
+        this.filterTimeFrame = hash.timeframe;
```

Comment:
This is not enough. Previously, `commit` and `day` granularity have the same zoom-in panel. Now, the zoom-in pannel for 'week' is different. When the user firstly opens a zoom-in pannel with `day`/`commit` granularity and then switch to 'week' granularity. The zoom-in pannel should react to this action. (Also change to `week` granularity)

Reply:
So for `commit` and `day` granularity, they will have the same ramp (ramp slices are in terms of `commit`) in the zoom-in panel. But for `week` granularity, the ramp in the zoom-in panel should have ramp slices that are in terms of commits for the entire `week`?

Reply:
Or should it be: 
- `commit` granularity will have ramp slices in terms of `commit` in the zoom-in panel
- `day` granularity will have ramp slices in terms of `day` in the zoom-in panel
- `week` granularity will have ramp slices in terms of `week` in the zoom-in panel

Reply:
@fzdy1914

Reply:
Sorry for not seeing the reply. Maybe I am not clear enough. See the following pics.
Now:
![a](https://user-images.githubusercontent.com/35621726/59286074-a6593c00-8ca1-11e9-9740-0d26818caca4.gif)
Expect:
![image](https://user-images.githubusercontent.com/35621726/59286090-ae18e080-8ca1-11e9-9e53-d41579fbda48.png)

Reply:
After changing the granularity to `week`, the right panel should also be the one for the week.

Reply:
Oh you mean for the zoom panel to change accordingly when switching the `granularity`. May I address this as a separate PR as it might not be suitable to address this issue in this PR?

Reply:
> Or should it be:
> 
> * `commit` granularity will have ramp slices in terms of `commit` in the zoom-in panel
> * `day` granularity will have ramp slices in terms of `day` in the zoom-in panel
> * `week` granularity will have ramp slices in terms of `week` in the zoom-in panel

@fzdy1914 
Also, shouldn't the zoom-in panel perform as described above? Because the current implementation of zoom-in feature is basically showing the zoomed-in ramp slices in terms of `commit` no matter granularity is set to `commit`, `day` or `week`, which might be redundant as they show the same ramp.

Reply:
>May I address this as a separate PR as it might not be suitable to address this issue in this PR?

I think it is better to do it in this PR. Firstly, it is considered as a whole feature. Meanwhile, it is not so difficult to implement.

Reply:
> > Or should it be:
> > 
> > * `commit` granularity will have ramp slices in terms of `commit` in the zoom-in panel
> > * `day` granularity will have ramp slices in terms of `day` in the zoom-in panel
> > * `week` granularity will have ramp slices in terms of `week` in the zoom-in panel
> 
> @fzdy1914
> Also, shouldn't the zoom-in panel perform as described above? Because the current implementation of zoom-in feature is basically showing the zoomed-in ramp slices in terms of `commit` no matter granularity is set to `commit`, `day` or `week`, which might be redundant as they all show the same ramp.

I cannot decide which one is better. We can ask @damithc's opinion of this zoom-in panel.

Reply:
I think the zoom feature was only meant for commit granularity. For other granularities, there is no risk of having too many ramps on top of each other.

Reply:
I think there are still cases where many ramps on top of each other when granularity is set to `day` and `week`. For example:
<img width="818" alt="Screen Shot 2019-06-13 at 12 29 47 PM" src="https://user-images.githubusercontent.com/32864116/59403946-567c9100-8dd7-11e9-8c83-65208403fa45.png">
<img width="813" alt="Screen Shot 2019-06-13 at 12 29 54 PM" src="https://user-images.githubusercontent.com/32864116/59403949-57adbe00-8dd7-11e9-8cb0-18345fd28c08.png">

Reply:
> I think there are still cases where many ramps are on top of each other when granularity is set to `day` and `week`.

Yes, if the analysis is over a long period. We can support zoom for other granularity too if that is not too much extra work, but it is not that urgent as our immediate use case is analysis of student work over a semester.

Reply:
I have modified it to support zoom for other granularities.